src/utils/statuses/groupStatusesByUmbrellaStatus.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  taskStatusToCopy,
+  mapTaskStatusToUmbrellaStatus,
+  mapUmbrellaStatusColors,
+  sortedUmbrellaStatus,
+} from "constants/task";
+
+type ColorCount = {
+  count: number;
+  statuses: string[];
+  color: string;
+  umbrellaStatus: string;
+};
+
+export const groupStatusesByUmbrellaStatus = (
+  statusCounts: { status: string; count: number }[]
+) => {
+  const counts: { [key: string]: ColorCount } = {};
+
+  statusCounts.forEach((stat) => {
+    const umbrellaStatus = mapTaskStatusToUmbrellaStatus[stat.status];
+    if (counts[umbrellaStatus]) {
+      counts[umbrellaStatus].count += stat.count;
+      if (!counts[umbrellaStatus].statuses.includes(stat.status)) {
+        counts[umbrellaStatus].statuses.push(taskStatusToCopy[stat.status]);
+      }
+    } else {
+      counts[umbrellaStatus] = {
+        count: stat.count,
+        statuses: [taskStatusToCopy[stat.status]],
+        color: mapUmbrellaStatusColors[umbrellaStatus].border,
```

Comment:
This color resembles commit chart tooltip icon fill and barchart fill. We may need to expand a different value from mapUmbrellaStatusColors in here depending on design feedback.